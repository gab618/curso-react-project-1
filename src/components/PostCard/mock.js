export const postCardPropsMock = {
  id: 1,
  body: "body 1",
  title: "Meu t√≠tulo massa",
  cover: "https://github.com/gab618.png",
};
