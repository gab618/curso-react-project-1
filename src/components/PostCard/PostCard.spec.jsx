import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardPropsMock } from "./mock";

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard post={postCardPropsMock} />);

    expect(
      screen.getByRole("img", { name: postCardPropsMock.title })
    ).toHaveAttribute("src", postCardPropsMock.cover);

    expect(
      screen.getByRole("heading", { name: postCardPropsMock.title })
    ).toBeInTheDocument();

    expect(screen.getByText(postCardPropsMock.body)).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard post={postCardPropsMock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
