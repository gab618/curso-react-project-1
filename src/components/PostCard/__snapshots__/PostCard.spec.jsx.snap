// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Meu título massa"
    src="https://github.com/gab618.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Meu título massa
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
